src/ochre_simulation.py

```diff
@@ -0,0 +1,104 @@
+import os
+import xmltodict
+import click
+from datetime import datetime, timedelta
+import shutil
+from ochre import Dwelling
+
+"""
+Code for running simulations on OCHRE.
+
+Author: Alex Lee (alexlee5124@gmail.com)
+Date: 4/19/2025
+
+"""
+
+def extract_weather_station(file_path):
+    try:
+        with open(file_path, 'r', encoding='utf-8') as file:
+            buildingXML = xmltodict.parse(file.read())
+            weather_station_name = buildingXML['HPXML']['Building']['BuildingDetails']['ClimateandRiskZones']['WeatherStation']['Name']
+    except Exception as e:
+        print(f"Error parsing XML {file_path}: {e}")
+
+    return weather_station_name
+
+@click.command()
+@click.argument('building_id', type=int)
+@click.argument('upgrade_id', type=int)
+@click.argument('state', type=str)
+def simulate_dwelling(
+        building_id:int,
+        upgrade_id:int,
+        state:str,
+        year=2022,
+        version="resstock_amy2018_release_1.1",
+        start_time=datetime(2007, 1, 1, 0, 0),
+        time_res=timedelta(minutes=60),
+        duration=timedelta(days=3),
+        ):
+    """
+    simulate: This function runs an OCHRE simulation for a building. It filters the building by the
+    building id, upgrade id, state, year, and version of the building XML file. You need to have already
+    downloaded the building XML file prior to running this function.
+
+    Input simulation parameters, such as duration, start date, simulation timestep size, etc.
+
+    Author: Alex Lee (alexlee5124@gmail.com)
+    Date: 3/29/2025
+    """
+
+    building_folder = f"bldg{building_id:07}-up{upgrade_id:02}"
+    data_folder_path = os.path.join(os.path.dirname(os.path.dirname(os.getcwd())), "data")
+
+    input_filepath = os.path.join(data_folder_path, "/building_energy_models/"
+                                  f"{state}_{year}_{version}/{building_folder}")
+    # Load XML and schedule file
+    xml_file = None
+    schedule_file = None
+    for file in os.listdir(input_filepath):
+        if file.endswith(".xml"):
+            xml_file = os.path.join(input_filepath, file)
+            # Extract weather station name if XML file exists
+            weather_station = extract_weather_station(xml_file)
+            weather_file=os.path.join(data_folder_path,f"weather/BuildStock_TMY3_FIPS/{weather_station}.epw")
+        elif file.endswith(".csv"):
+            schedule_file = os.path.join(input_filepath, file)
+    if not xml_file:
+        print(f"Building XML file does not exist for: bldg{building_id:07}-up{upgrade_id:02}")
```

Comment:
Move this to the error handling I described in the previous comment: if the file isn't found, print the files path and exit the entire script with an error code, don't return like this.